lib/FrontendTool/FrontendTool.cpp

```diff
@@ -279,7 +279,8 @@ static void computeSwiftModuleTraceInfo(
     path::replace_extension(modPath, swiftInterfaceExt);
   };
 
-  for (auto &depPath : depTracker.getDependencies()) {
+  for (auto &dp : depTracker.getDependencies()) {
+    auto depPath = llvm::sys::path::convert_to_slash(dp);
```

Comment:
Shouldn't the fix involve us normalizing paths when we _add_ them to the dependency tracker (if not even before that), not normalizing them when we remove them? Otherwise, every caller of the dependency tracker needs to be aware of this gotcha'.

Reply:
My intention was to disrupt as less as possible the existing code. Since the comparison was happening with the values extracted from `getDependencies()` and from the map prepared below, I decided to do the change here, and ensure nothing else should change (which might create more problems).

If you point me to the point the depenency paths are added to the `depTracker` and to the `ModuleDecl` of the loaded modules, I can figure out how to change it (and hopefully it will not break anything else).

Reply:
There is a method `Dependency::addDependency` which I think should be responsible for normalizing the paths. cc @jrose-apple for a second opinion.

In case, that does creates further issues, it's probably not worth it to block CI on this, so you could commit the solution you have right now (normalize paths when extracting them from the depTracker) and file a JIRA for a cleanup task (cc/assign it to me).

Reply:
I don't like the dependency tracker normalizing paths because in theory a Makefile ought to use the path style of the current system. I guess it could normalize *back* and that would help with mixes, but…

(Also we don't control all the inputs to the dependency tracker; some of them come from Clang.)

Reply:
When I say "normalizing", I meant doing so in a system-sensitive way, not in a system-agnostic way (tbf, I should've been more clear). I don't know what `llvm::sys::path` functions do w.r.t. normalization because it isn't documented :-/. For example, all the examples for `append` use slashes, not backslashes. Uhhhhh....